torch/onnx/_internal/exporter.py

```diff
@@ -350,16 +351,31 @@ class ResolvedExportOptions(ExportOptions):
 
     @_beartype.beartype
     def __init__(
-        self, options: Optional[Union[ExportOptions, "ResolvedExportOptions"]]
+        self,
+        options: Optional[Union[ExportOptions, "ResolvedExportOptions"]] = None,
+        model: Optional[Union[torch.nn.Module, Callable, torch_export.ExportedProgram]] = None,  # type: ignore[name-defined]
```

Comment:
This I would like to pick your brain. Adding `model` solely for validation purposes. Should I move the validation outside `ResolvedExportOptions` and let the API simpler?